lib/src/main/java/grpcbridge/parser/ProtoFormDataParser.java

```diff
@@ -0,0 +1,125 @@
+package grpcbridge.parser;
+
+import com.google.common.base.Function;
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Message;
+import com.google.protobuf.util.JsonFormat;
+import grpcbridge.Exceptions;
+import grpcbridge.http.HttpResponse;
+import grpcbridge.route.Route;
+import grpcbridge.rpc.RpcMessage;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.UnsupportedEncodingException;
+import java.lang.reflect.Type;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+
+public final class ProtoFormDataParser extends ProtoParser {
+
+    private static final Gson gson = new Gson();
+    private static final Type type = new TypeToken<Map<String, String>>() {}.getType();
+    private static final String URL_ENCODED_FORM = "application/x-www-form-urlencoded";
+    public static final ProtoFormDataParser INSTANCE = new ProtoFormDataParser();
+
+    private ProtoFormDataParser() {}
+
+    @Override
+    public String serialize(
+        Integer index,
+        JsonFormat.Printer printer,
+        @Nonnull Message message
+    ) {
+        try {
+            String json = printer.print(message);
+            Map<String, String> msg = new Gson().fromJson(json, type);
+            msg = msg.entrySet().parallelStream().map(it -> {
```

Comment:
The convention is:

```
something.stream()
    .map(...)
    .collect(...)
```
Otherwise it is hard to see that collect applies to the map.